modules/server_new/src/main/scala/observe/server/gmos/GmosSouth.scala

```diff
@@ -74,23 +68,62 @@ object GmosSouth {
         DynamicConfig.GmosSouth.roi.asGetter
       override val readout: Getter[DynamicConfig.GmosSouth, GmosCcdMode]                          =
         DynamicConfig.GmosSouth.readout.asGetter
+      override val isMosPreimaging: Getter[StaticConfig.GmosSouth, MosPreImaging]                 =
+        StaticConfig.GmosSouth.mosPreImaging.asGetter
+    }
 
   def build[F[_]: Temporal: Logger](
     controller:        GmosController[F, GmosSite.South.type],
     dhsClientProvider: DhsClientProvider[F],
     nsCmdR:            Ref[F, Option[NSObserveCommand]],
-    obsType:           StepType,
+    stepType:          StepType,
     staticCfg:         StaticConfig.GmosSouth,
     dynamicCfg:        DynamicConfig.GmosSouth
-  ): Either[ObserveFailure, GmosSouth[F]] =
-    Gmos
-      .buildConfig[F, GmosSite.South.type, StaticConfig.GmosSouth, DynamicConfig.GmosSouth](
-        Instrument.GmosS,
-        obsType,
-        staticCfg,
-        dynamicCfg
+  ): GmosSouth[F] = GmosSouth(
+    controller,
+    dhsClientProvider,
+    nsCmdR,
+    Gmos.buildConfig[F, GmosSite.South.type, StaticConfig.GmosSouth, DynamicConfig.GmosSouth](
+      Instrument.GmosS,
+      stepType,
+      staticCfg,
+      dynamicCfg
+    )
+  )
+
+  def obsKeywordsReader[F[_]: MonadThrow](
+    staticConfig:  StaticConfig.GmosSouth,
+    dynamicConfig: DynamicConfig.GmosSouth
+  )(using
+    getters:       Gmos.ParamGetters[GmosSite.South.type, StaticConfig.GmosSouth, DynamicConfig.GmosSouth]
+  ): GmosObsKeywordsReader[F,
+                           GmosSite.South.type,
+                           StaticConfig.GmosSouth,
+                           DynamicConfig.GmosSouth
+  ] =
+    GmosObsKeywordsReader(staticConfig, dynamicConfig)
+
+  object specifics extends InstrumentSpecifics[StaticConfig.GmosSouth, DynamicConfig.GmosSouth] {
+    override val instrument: Instrument = Instrument.GmosS
+
+    override def calcStepType(
+      stepConfig:   StepConfig,
+      staticConfig: StaticConfig.GmosSouth,
+      instConfig:   DynamicConfig.GmosSouth,
+      obsClass:     ObserveClass
+    ): Either[ObserveFailure, StepType] =
+      Gmos.calcStepType(instrument,
+                        stepConfig,
+                        staticConfig,
+                        obsClass,
+                        gsParamGetters.nodAndShuffle
       )
-      .map { case (t, config) =>
-        GmosSouth(controller, dhsClientProvider, nsCmdR, t, config)
-      }
+
+    override def sfName(config: DynamicConfig.GmosSouth): LightSinkName = LightSinkName.Gmos
+
+    // TODO Use different value if using electronic offsets
+    override val oiOffsetGuideThreshold: Option[Length] =
+      (Arcseconds(0.01) / FOCAL_PLANE_SCALE).some
```

Comment:
We should drop squants and move to coulomb. For another PR

Reply:
I did it for `Time`. `Angle` and `Length` come next, but it will be a bit more complicated. Specially because of the use that `FOCAL_PLANE_SCALE`